feature/xhr/aggregate/deny-list.js

```diff
@@ -0,0 +1,80 @@
+
+module.exports = {
+  shouldCollectEvent: shouldCollectEvent,
+  setDenyList: setDenyList
+}
+
+var denyList = []
+
+function shouldCollectEvent(params) {
+  if (denyList.length === 0) {
+    return true
+  }
+
+  for (var i = 0; i < denyList.length; i++) {
+    var parsed = denyList[i]
+    if (parsed.hostname === '*') {
+      return false
+    }
+    if (compareDomain(parsed.hostname, params.hostname) &&
+        comparePath(parsed.pathname, params.pathname)) {
+      return false
+    }
+  }
+  return true
+}
+
+function setDenyList(denyListConfig) {
+  denyList = []
+  if (!denyListConfig || !denyListConfig.length) {
+    return
+  }
+  for (var i = 0; i < denyListConfig.length; i++) {
+    var url = denyListConfig[i]
+    if (url.indexOf('http://') === 0) {
+      url = url.substring(7)
```

Comment:
Just to be aware, there are some web applications that use a URL pattern that includes two or more cases of http(s):// in the string

If the url starts with `https://` but includes `http://` in the rest of the url, it could in theory cause an unexpected substring from the `url.substring(7)`

[example link](https://web.archive.org/web/20101224003257/http://webmasters.stackexchange.com/)

Reply:
I think it would still be ok since the substring call is removing just the protocol at the very beginning of the URL. If there was a second protocol string, it would be in the path (as in the example). I have added a test that covers this. Let me know if there is something I am missing.